static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx

```diff
@@ -93,74 +81,109 @@ function WidgetBuilder({
     });
   }
 
-  if (!dataSet) {
-    return (
-      <Alert type="error" icon={<IconWarning />}>
-        {t('Data set not found.')}
-      </Alert>
-    );
-  }
-
-  function handleAddWidget(newWidget: Widget) {
-    onSave([...dashboard.widgets, newWidget]);
-  }
-
-  if (
-    (isEditing && !defined(widgetId)) ||
-    (isEditing && defined(widgetId) && !dashboard.widgets[widgetId])
-  ) {
-    return (
-      <Alert type="error" icon={<IconWarning />}>
-        {t('Widget not found.')}
-      </Alert>
-    );
-  }
-
-  function handleUpdateWidget(nextWidget: Widget) {
-    if (!widgetId) {
-      return;
-    }
-
-    const nextList = [...dashboard.widgets];
-    nextList[widgetId] = nextWidget;
-    onSave(nextList);
-  }
-
-  function handleDeleteWidget() {
-    if (!widgetId) {
-      return;
-    }
-    const nextList = [...dashboard.widgets];
-    nextList.splice(widgetId, 1);
-    onSave(nextList);
-  }
-
-  if (dataSet === DataSet.EVENTS) {
-    return (
-      <EventWidget
-        dashboardTitle={dashboard.title}
-        widget={widget}
-        onAdd={handleAddWidget}
-        onUpdate={handleUpdateWidget}
-        onDelete={handleDeleteWidget}
-        onChangeDataSet={handleDataSetChange}
-        goBackLocation={goBackLocation}
-        isEditing={isEditing}
-      />
-    );
-  }
+  // function handleAddWidget(newWidget: Widget) {
+  //   onSave([...dashboard.widgets, newWidget]);
+  // }
+
+  // function handleUpdateWidget(nextWidget: Widget) {
+  //   if (!widgetId) {
+  //     return;
+  //   }
+
+  //   const nextList = [...dashboard.widgets];
+  //   nextList[widgetId] = nextWidget;
+  //   onSave(nextList);
+  // }
+
+  // function handleDeleteWidget() {
+  //   if (!widgetId) {
+  //     return;
+  //   }
+  //   const nextList = [...dashboard.widgets];
+  //   nextList.splice(widgetId, 1);
+  //   onSave(nextList);
+  // }
 
   return (
-    <MetricWidget
-      organization={organization}
-      router={router}
-      location={location}
-      dashboardTitle={dashboard.title}
-      params={params}
-      goBackLocation={goBackLocation}
-      onChangeDataSet={handleDataSetChange}
-    />
+    <SentryDocumentTitle title={dashboard.title} orgSlug={orgSlug}>
+      {!Object.values(DataSet).includes(dataSet) ? (
```

Comment:
I usually like the approach you suggested better too and try to avoid using ternaries, but recently I reviewed some PRs and didn't find it hard to read, so I decided to use it here too...but I can update the code

Reply:
Regarding the `SentryDocumentTitle `..yes this could be done and it would make the code simpler but I think I prefer to avoid touching the index file as it is a temporary file that will be removed when we remove the feature flag. When this happens we can only rename the `widgetBuilder.tsx `to be the `index.tsx`

Reply:
we could also return the `SentryDocumentTitle` and create a function for the content, for instance, `renderContent()` 🤷‍♀️